common/test/acceptance/tests/lms/test_crowdsourcehinter_problem.py

```diff
@@ -0,0 +1,92 @@
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        # Install a course with sections/problems, tabs, updates, and handouts
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+        problem_data = dedent('''
+            <problem>
+                <p>A text input problem accepts a line of text from the student, and evaluates the input for correctness based on an expected answer.</p>
+                <p>The answer is correct if it matches every character of the expected answer. This can be a problem with international spelling, dates, or anything where the format of the answer is not clear.</p>
+                <p>Which US state has Lansing as its capital?</p>
+                <stringresponse answer="Michigan" type="ci" >
+                      <textline label="Which US state has Lansing as its capital?" size="20"/>
+                </stringresponse>
+                <solution>
+                <div class="detailed-solution">
+                <p>Explanation</p>
+                <p>Lansing is the capital of Michigan, although it is not Michigan's largest city, or even the seat of the county in which it resides.</p>
+                </div>
+                </solution>
+            </problem>
+        ''')
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection').add_children(
+                    XBlockFixtureDesc('vertical', 'Test Unit').add_children(
+                        XBlockFixtureDesc('problem', 'text input problem', data=problem_data), XBlockFixtureDesc('crowdsourcehinter', 'test crowdsourcehinter')
+                    )
+                )
+            )
+        ).install()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(self.browser, username=self.USERNAME, email=self.EMAIL,
+                     course_id=self.course_id, staff=False).visit()
+
+    def _goto_csh_problem_page(self):
+        self.courseware_page.visit()
+        csh_problem_page = CrowdsourcehinterProblemPage(self.browser)
+        self.assertTrue(len(self.browser.find_elements_by_class_name('crowdsourcehinter_block')) > 0)
+        return csh_problem_page
+
+    def test_student_hint_workflow(self):
+        """
+        Test the basic workflow of a student recieving hints. The student should submit an incorrect answer and
+        receive a hint (in this case no hint since none are set), be able to rate that hint, see a different UX
+        after submitting a correct answer, and be capable of contributing a new hint to the system.
+        """
+        csh_problem_page = self._goto_csh_problem_page()
+
+        csh_problem_page.submit_text_answer("michigann")
+        csh_problem_page.wait_for_ajax()
+        time.sleep(20)
```

Comment:
Do we really need these sleeps?  This one test will take a minute to execute.  We are trying to shave time from the test suite.